test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt

```diff
@@ -0,0 +1,152 @@
+import androidx.appcompat.app.AppCompatActivity
+import androidx.webkit.WebViewAssetLoader
+import androidx.webkit.WebViewClientCompat
+import com.bentley.itwin.IModelJsHost
+import com.bentley.itwin.MobileFrontend
+import java.io.IOException
+import java.io.InputStream
+import java.net.URLConnection
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+
+typealias PickUriContractType = ActivityResultContract<Nothing?, Uri?>
+
+class PickUriContract : PickUriContractType() {
+    private lateinit var context: Context
+
+    override fun createIntent(context: Context, input: Nothing?): Intent {
+        this.context = context
+        return Intent()
+            .setAction(Intent.ACTION_OPEN_DOCUMENT)
+            .setType("*/*")
+            .addCategory(Intent.CATEGORY_OPENABLE)
+    }
+
+    private fun getDisplayName(uri: Uri): String {
+        return FileHelper.getFileDisplayName(uri, context.contentResolver) ?: "unknownDisplayName"
+    }
+
+    override fun parseResult(resultCode: Int, intent: Intent?): Uri? {
+        val uri = intent?.takeIf { resultCode == Activity.RESULT_OK }?.data
+        if (uri != null) {
+            val destDir = "bim_cache"
+            FileHelper.copyToExternalFiles(context, uri, destDir, getDisplayName(uri))?.let { result ->
+                return Uri.parse(result)
+            }
+        }
+        return uri
+    }
+}
+
+// Very similar to WebViewAssetLoader.AssetsPathHandler except it doesn't log an error when a resource isn't found, also doesn't handle SvgZ streams.
+class AssetsPathHandler(context: Context) : WebViewAssetLoader.PathHandler {
+    private val mContext = context
+
+    private fun removeLeadingSlash(path: String): String {
+        return if (path.startsWith('/')) path.substring(1) else path
+    }
+
+    private fun openAsset(path: String): InputStream {
+        return mContext.assets.open(removeLeadingSlash(path), AssetManager.ACCESS_STREAMING)
+    }
+
+    private fun guessMimeType(path: String): String {
+        return URLConnection.guessContentTypeFromName(path) ?: "text/plain"
+    }
+
+    override fun handle(path: String): WebResourceResponse? {
+        return try {
+            WebResourceResponse(guessMimeType(path), null, openAsset(path))
+        } catch (e: IOException) {
+            WebResourceResponse(null, null, null)
+        }
+    }
+}
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var host: IModelJsHost
+    private var promiseName: String = ""
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        WebView.setWebContentsDebuggingEnabled(true)
+        val alwaysExtractAssets = true // for debugging, otherwise the host will only extract when app version changes
+        host = IModelJsHost(this, alwaysExtractAssets, true)
+        host.startup()
+        val frontend: MobileFrontend = object : MobileFrontend(host, "&standalone=true") {
```

Comment:
I would remove the type for the variable on the left of the equals sign:

```kotlin
        val frontend = object : MobileFrontend(host, "&standalone=true") {
```

Reply:
Fixed